cmd/nsdns/watch.go

```diff
@@ -35,50 +32,35 @@ func watchCommand() *cobra.Command {
 			}
 
 			dm.RefreshesCacheOnUpdate = true
-			if err := dm.UpdateCache(); err != nil {
+
+			clientset, err := GetKubernetesClientSet()
+			if err != nil {
 				return err
 			}
 
+			// Before starting the informer, start up a loop to update caches
+			// safely.
+			done := make(chan struct{})
```

Comment:
Why bother with this at all? The only thing that's done from the time this goroutine is started so the time the done channel allows flow to continue is the creation of the `informerFactory`. Seems like extra complexity just to have extra complexity.